### ISSUE TITLE
always_null and others inside view

### ISSUE THREAD

**Issue by ethey** [NLnetLabs/unbound#474]
This is regarding issue the feature request #397 and the implementation of the `always_null` zone.

The `always_null` zone does not seem to work inside a view:
```
server:
  access-control-view: 127.0.0.0/8 "local"

view:
  name: "local"
  local-zone: "example.com" always_null
```
gives the error `error: local-zone type: expected static, deny, refuse, redirect, transparent, typetransparent, inform, inform_deny, always_transparent, always_refuse, always_nxdomain, noview or nodefault`, while there is no problem if the local zone is in the `server` section.

The documentation indicates that the local-zone inside a view should `Ha[ve] the same types and behaviour  as  the global local-zone elements`, however it is not implemented in [configparser.y](https://github.com/NLnetLabs/unbound/blob/77451862f0455c532aa9f6d30f8fb429d2f3a609/util/configparser.y#L2830). `always_nodata` and `always_deny` also seem to be missing.

I have tested version 1.13.1 on Debian and Arch.

**Comment by wcawijngaards** [comment]
Thanks for the report. The checks for the input was missing on these types, fixed the checks and the always_null, always_nodata and always_deny local zone types work in a view local-zone.

**Comment by pupto-bismol** [comment]
Just ran into this myself today, it looks as if it doesn't still doesn't apply for `block_a` (I imagine no one uses this type much so it just hasn't come up until now)

```
server:
    access-control-view: 2001:db8:1234:5678::/64 "ipv6-only"

view:
    name: "ipv6-only"
    local-zone: example.com block_a
```

results in `error: local-zone type: expected static, deny, refuse, redirect, transparent, typetransparent, inform, inform_deny, inform_redirect, always_transparent, always_refuse, always_nxdomain, always_nodata, always_deny, always_null, noview, nodefault or ipset`. Works fine when using `local-zone: example.com block_a` in the `server` section.

Running on OpenWrt 24.10.5, unbound version 1.25.1

As a side note, what I wanted to use this for (filtering ipv4 records for subnets with no legacy addressing) can be solved just fine with tags I believe, and 79e100a7fbb45bee3a409f2add927dbee563d35d also helps quite a bit - so this isn't a giant concern - but still a bit of a footgun in the documentation. Also potentially worth making sure that the recent `block_a_wdata`/`block_aaaa_wdata` additions aren't missed either.

**Comment by wcawijngaards** [comment]
The commit https://github.com/NLnetLabs/unbound/commit/a6ccb2f74daad494565cff7fcf284f0ac5e81485 fixes so that `block_a` and  `block_a_wdata`, `block_aaaa_wdata` and so, are also allowed inside a view. The view error check was not up to date with the new type addition, and the fix adds it, so that the local-zone type is allowed there.
